src/ints/bios_disk.cpp

```diff
@@ -99,34 +100,33 @@ void incrementFDD(void) {
 	CMOS_SetRegister(0x14, (Bit8u)(equipment&0xff));
 }
 
-void swapInDisks(void) {
-	bool allNull = true;
-	Bit32s diskcount = 0;
-	Bit32s swapPos = swapPosition;
-	Bit32s i;
-
-	/* Check to make sure that  there is at least one setup image */
-	for(i=0;i<MAX_SWAPPABLE_DISKS;i++) {
-		if(diskSwap[i]) {
-			allNull = false;
+template<typename T, size_t N>
+static size_t disk_array_prefix_size(const std::array<T, N> &images) {
```

Comment:
A more readable replacement with a better name.

"swapInDisks" sounds like it should return a bool... if a "swap" (whatever that is) exists in the "Disks".

Reply:
I went back and forth on naming of this function :/. One name I tried earlier was `boot_floppy_num`, but it would be incorrect if array containing floppies had null pointers in the middle. I think for `diskSwap` it never happens, but definitely happens for the other array.

Reply:
`disk_array_prefix_size`, the 'prefix size' portion perfectly describes the function, because you only count the first N valid items until a nullptr is hit, and return that quantity - even if there are more valid array elements after the first nullptr.  So I would say you nailed it. 

(If prefix, or first, or initial, were not in the name, I would have mentioned something about not being a complete count of items - even though that was the intent).